packages/client/src/open-feature.ts

```diff
@@ -81,16 +86,22 @@ export class OpenFeatureAPI extends OpenFeatureCommonAPI<Provider, Hook> impleme
       const oldContext = this._context;
       this._context = context;
 
-      const providersWithoutContextOverride = Array.from(this._clientProviders.entries())
+      // collect all providers that are using the default context (not mapped to a name)
+      const defaultContextNameProviders: NameProviderTuple[] = Array.from(this._clientProviders.entries())
         .filter(([name]) => !this._namedProviderContext.has(name))
-        .reduce<Provider[]>((acc, [, provider]) => {
-          acc.push(provider);
+        .reduce<{ namespace: string; provider: Provider }[]>((acc, [name, provider]) => {
```

Comment:
Reuse the type above once we have a better name.

Reply:
Ya, that was why I created the name above.